packages/core/src/components/BlockManager.ts

```diff
@@ -177,7 +177,12 @@ export class BlocksManager {
     try {
       const blockElement = await block.render();
 
-      this.#editorUI.addBlock(blockElement, index.blockIndex);
+      this.#eventBus.dispatchEvent(new BlockAddedCoreEvent({
+        tool: tool.name,
+        data: data.data,
+        uiContent: blockElement,
```

Comment:
maybe `element`?

Reply:
It could be not only HTMLElement, basically anything the tool returns